packages/core/admin/ee/admin/content-manager/components/Filter/CustomInputs/ReviewWorkflows/AssigneeFilter.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import { Combobox, ComboboxOption } from '@strapi/design-system';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+
+import { getDisplayName } from '../../../../../../../admin/src/content-manager/utils/getDisplayName';
+import { useAdminUsers } from '../../../../../../../admin/src/hooks/useAdminUsers';
+
+export const AssigneeFilter = ({ value, onChange }) => {
+  const { formatMessage } = useIntl();
+  const { users, isLoading } = useAdminUsers();
+
+  return (
+    <Combobox
+      value={value}
+      aria-label={formatMessage({
+        id: 'content-manager.components.Filters.usersSelect.label',
+        defaultMessage: 'Search and select an user to filter',
+      })}
+      onChange={onChange}
+      loading={isLoading}
+    >
+      {users.map((user) => {
+        return (
+          <ComboboxOption key={user.id} value={user.id.toString()}>
+            {getDisplayName(user, formatMessage)}
+          </ComboboxOption>
+        );
+      })}
```

Comment:
going to QA during the day but just in case you know, how does this behave if you don't have permissions to see users?

Reply:
Good point. That is actually a problem with creator fields too, @Feranchz. Should we render a no-permission component in these cases?

Reply:
@Feranchz I have changed the current implementation a bit, because if I am testing this with a user that doesn't have permissions to read users we are still fetching the list of users in the background:

https://github.com/strapi/strapi/blob/3cb214d2739b90217b36a1916e2d8bb0d129a260/packages/core/admin/admin/src/content-manager/pages/ListView/index.js#L136-L150

I have used the same logic to determine whether the assignee filter should be displayed or not https://github.com/strapi/strapi/blob/3cb214d2739b90217b36a1916e2d8bb0d129a260/packages/core/admin/admin/src/content-manager/pages/ListView/index.js#L246-L262


I think for a first step that makes more sense, but I will confirm that with Luccas once he is back.

Ok for everyone?